daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-publish-subscribe.md

```diff
@@ -284,8 +289,39 @@ To publish a message to a topic, invoke the following endpoint on a Dapr instanc
 {{< tabs "Dapr CLI" "HTTP API (Bash)" "HTTP API (PowerShell)">}}
 
 {{% codetab %}}
+To use the Dapr CLI, you need to have an application running that has the permissions to publish topics. Create a new file called `pubapp.js` with the following code.
```

Comment:
Why is this application needed? My understanding was the CLI publishes straight to the topic without needing an application. Looking at this example it seems like we are creating an app that converts service invocation requests to publish events. @yaron2 @msfussell

Reply:
Yes, this changed in the CLI, since used to mean that there was a security hole in the CLI, that could get around the permissions for publishing, since you can restrict who published. If you look at the pub/sub sample here https://github.com/dapr/quickstarts/tree/master/pub-sub#use-the-cli-to-publish-messages-to-subscribers we do the same approach now with the React app publishing messages. Here is the CLI https://github.com/dapr/cli#publishsubscribe.

Reply:
Ok I'm confused.

- If the app in this example is just to open up permissions why is it not blank? Why does it contain logic to publish data?
- It seems like the publish cli command is useless if all it's doing is invoking a service to then publish from the app.
- In this example you've given, what is the pubsub cli command doing the publishing or the app? Either we need to remove the code in the app or get rid of the cli publish command if all it can do is invoke